src/WhatToCook.WebApp/ClientApp/src/app/recipes/recipe-list/recipe-list.component.ts

```diff
@@ -1,62 +1,33 @@
-import { Component, OnInit } from '@angular/core';
-import { RecipeCardViewModel } from '../recipe-card/recipe-card.component';
+import {Component, Input, OnInit} from '@angular/core';
+import {RecipeService} from '../recipe.service';
+import {Recipe} from '../Recipe';
+import {CreateRecipe} from '../recipe-view/CreateRecipe';
+
+export type filterRecipePredicate = (recipe: Recipe) => boolean;
+
 @Component({
   selector: 'app-recipe-list',
   templateUrl: './recipe-list.component.html',
   styleUrls: ['./recipe-list.component.scss']
 })
 export class RecipeListComponent implements OnInit {
-  recipeCards: RecipeCardViewModel[] = [{ 
-    name: 'Flaczki',
-    description: "Opis flaczków",
-    img: "/assets/images/Tomato.jpg"
-  },
-  {
-    name: 'Flaczki 2',
-    description: "Opis flaczków 2",
-    img: "/assets/images/Tomato.jpg"
-  },
-{
-  name: 'Flaczki 3',
-  description: "Opis flaczków 2",
-  img: "/assets/images/Tomato.jpg"
-},
-{
-  name: 'Flaczki 4',
-  description: "Opis flaczków 2",
-  img: "/assets/images/Tomato.jpg"
-},
-{
-  name: 'Flaczki 5',
-  description: "Opis flaczków 2",
-  img: "/assets/images/Tomato.jpg"
-},
-{
-  name: 'Flaczki 6',
-  description: "Opis flaczków 2",
-  img: "/assets/images/Tomato.jpg"
-},
-{
-  name: 'Flaczki 7',
-  description: "Opis flaczków 2",
-  img: "/assets/images/Tomato.jpg"
-},
-{
-  name: 'Flaczki 8',
-  description: "Opis flaczków 2",
-  img: "/assets/images/Tomato.jpg"
-},
-{
-  name: 'Flaczki 9',
-  description: "Opis flaczków 2",
-  img: "/assets/images/Tomato.jpg"
-},
-];
-   
-  constructor(){
+  recipes: Recipe[] = [];
+
+  @Input() filterMethod?: filterRecipePredicate;
+
+  constructor(private recipeService: RecipeService) {
   }
 
-  ngOnInit() {
-      //Tutaj pobieramy dane z API - możemy tu wykonywać operacje z async
+  ngOnInit(): void {
+    //Tutaj pobieramy dane z API - możemy tu wykonywać operacje z async
+    this.recipeService.get().subscribe(
+      (recipes) => {
```

Comment:
Use new syntax, see other comments how to do it

Reply:
done